src/main/java/com/android/volley/toolbox/DiskBasedCacheUtility.java

```diff
@@ -37,6 +43,95 @@ static String getFilenameForKey(String key) {
         return localFilename;
     }
 
+    /** Returns a file object for the given cache key. */
+    public static File getFileForKey(String key, FileSupplier rootDirectorySupplier) {
+        return new File(rootDirectorySupplier.get(), getFilenameForKey(key));
+    }
+
+    static boolean wouldExceedCacheSize(long newTotalSize, int maxCacheSize) {
+        return (newTotalSize >= maxCacheSize);
+    }
```

Comment:
I think it's safer to keep > rather than >= because that's what would have result in a prune, what do you think? (Same below)

Reply:
yeah, I think that makes sense. I'll change it back.

Reply:
I changed it in the method, but this lead to a test failing. In pruneIfNeeded, it was originally checking if totalSize <= maxCacheSize. To use the method here, I added 1 to totalSize, so to make the check the same as before.

Reply:
Should we update the test instead, by any chance?

Reply:
Agree. This code originated from

```
        if (mTotalSize + entry.data.length > mMaxCacheSizeInBytes
                && entry.data.length
                        > mMaxCacheSizeInBytes * DiskBasedCacheUtility.HYSTERESIS_FACTOR) {
            return;
        }
```

Reply:
`DiskBasedCache` had the following check in `#put`:

```java
        if (mTotalSize + entry.data.length > mMaxCacheSizeInBytes
                && entry.data.length
                        > mMaxCacheSizeInBytes * DiskBasedCacheUtility.HYSTERESIS_FACTOR) {
          return;
        }
```

Reply:
I don't think it super matters anyway. Since the check in put doesn't count the size of the CacheHeader (aside from the data itself). So if it was already really close to the max, it may get put and then pruned. I switched it to >= since that does make more sense (and then holds for the check in pruneIfNeeded), but I don't think it makes a big difference either way.

Reply:
Sounds good, thanks!